iothub_client/src/blob.c

```diff
@@ -224,12 +224,13 @@ BLOB_RESULT Blob_UploadMultipleBlocksFromSasUri(const char* SASURI, IOTHUB_CLIEN
                                     size_t size; /* source size set by getDataCallback */
 
                                     /* get first block */
-                                    getDataCallback(FILE_UPLOAD_OK, &source, &size, context);
+                                    IOTHUB_CLIENT_FILE_UPLOAD_GET_DATA_RESULT cbkRet;
+                                    cbkRet = getDataCallback(FILE_UPLOAD_OK, &source, &size, context);
 
                                     /*Codes_SRS_BLOB_99_002: [ If the size of the block returned by `getDataCallback` is 0 or if the data is NULL, then `Blob_UploadMultipleBlocksFromSasUri` shall exit the loop. ]*/
-                                    uploadOneMoreBlock = (source != NULL && size > 0) ? 1 : 0;
+                                    uploadOneMoreBlock = (cbkRet == IOTHUB_CLIENT_FILE_UPLOAD_GET_DATA_OK && source != NULL && size > 0) ? 1 : 0;
```

Comment:
I haven't elaborated on the logic myself, but since this return value is being checked a few times in the code, is there a way to re-write it so you have 1 if block?

Reply:
I redesigned the whole loop in a (hopefully) clearer and shorter way.